### PR TITLE
crimson: drop the support of ProtocolV1

### PR DESCRIPTION
**Pull request by rzarzynski**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by tchaikov — APPROVED** [review]

**Comment by tchaikov** [comment]
> When it enters the production, V1 will be long obsolete

I love to drop unnecessary piece of code. but can we pick the version of RADOS client ? what if user is still have ancient client which does not speak in v2 and wants to upgrade to a crimson cluster?

@liewegas @athanatos what do you think?

**Comment by athanatos** [comment]
@tchaikov v1 would be pretty old at that point.  We certainly can choose to cut support for very old versions of the rados client, we just haven't chose to do so yet so much.  Also, as I understand it, this support doesn't really work, so if we wanted to properly support those clients we could add it back in properly.

**Comment by tchaikov** [comment]
@athanatos if we've made the decision that we don't want to support v1, i'd be very happy to drop it. what i am worried about is

>  if we wanted to properly support those clients we could add it back in properly.

anyway. we'll see.

File: src/crimson/net/SocketMessenger.cc [hunk 1]
```diff
@@ -154,15 +155,18 @@ SocketMessenger::connect(const entity_addr_t& peer_addr, const entity_name_t& pe
   assert(seastar::this_shard_id() == master_sid);
 
   // make sure we connect to a valid peer_addr
-  ceph_assert(peer_addr.is_legacy() || peer_addr.is_msgr2());
+  if (!peer_addr.is_msgr2()) {
+    logger().warn("ProtocolV1 is no longer supported");
+    return nullptr;
```
Comment (cyx1231st) [inline src/crimson/net/SocketMessenger.cc:160]: I think `ceph_abort()` here is better. The current crimson OSD code doesn't expect or handle `nullptr` returned by `Messenger::connect()`. And if OSD need to deal with it, it should be done before calling `Messenger::connect()`.

**Review by cyx1231st — COMMENTED** [review]
v1 lossless policy and the compatibility to both v1 and v2 are not implemented in the current messenger. I'd be happy if we decide to drop v1 support in crimson.

Btw, also need to drop v1 tests in `test_messenger.cc`, such as `test_echo(rounds, keepalive_ratio, false)`, `test_concurrent_dispatch(false)` and `test_preemptive_shutdown(false)`.

File: src/crimson/net/SocketMessenger.cc (hunk 1 above)
Comment (rzarzynski) [reply src/crimson/net/SocketMessenger.cc]: Dropped.

**Comment by rzarzynski** [comment]
@tchaikov, @cyx1231st: addressed the comments and squashed fixups.

**Review by tchaikov — APPROVED** [review]

**Pull request merged**